src/misc/socket.cc

```diff
@@ -576,16 +580,25 @@ static ncclResult_t socketPollConnect(struct ncclSocket* sock) {
   if (ret == 0 || (ret < 0 && errno == EINTR)) {
     return ncclSuccess;
   } else if (ret < 0) {
-    WARN("socketPollConnect poll() failed with error %s", strerror(errno));
+    WARN(warnStr, ret, strerror(errno), pfd.revents, sockstr);
     return ncclRemoteError;
-  } else if (ret != 1 || (pfd.revents & POLLOUT) == 0) {
-    WARN("socketPollConnect poll() returned %d%s", ret, (pfd.revents & POLLOUT) ? "" : ", no POLLOUT events");
+  } else if (ret != 1) {
+    WARN(warnStr, ret, strerror(errno), pfd.revents, sockstr);
     return ncclSystemError;
   }
 
   /* check socket status */
-  SYSCHECK(getsockopt(sock->fd, SOL_SOCKET, SO_ERROR, (void*)&ret, &rlen), "getsockopt");
-  return socketConnectCheck(sock, ret, __func__);
+  SYSCHECK(getsockopt(sock->fd, SOL_SOCKET, SO_ERROR, (void*)&sockErrorCode, &errCodeLen), "getsockopt");
+  if ((pfd.revents & POLLOUT) == 0) {
+    /*
+     * When poll returns 1 however no POLLOUT, it could be POLLERR or POLLHUP.
+     * Those cases could be triggered by transient networking issue, so we would
+     * want to go to socketConnectCheck for retry, instead of returning error directly.
+     */
+    WARN(warnStr, ret, strerror(sockErrorCode), pfd.revents, sockstr);
```

Comment:
We can't call WARN then not error out. If we cover that case gracefully, then it should be an INFO.

Reply:
Done. Changed to INFO.